### PR TITLE
feat: Add functions reference document

### PR DESCRIPTION
**Pull request by HomelessDinosaur**

### CODE REVIEW

File: docs/reference/project/functions.mdx [hunk 1]
```diff
@@ -1 +1,46 @@
...
+nitric templates:repos:add function
+```
+
+Then add the function:
+
+```bash
+nitric make:service
+```
+
+This will populate the `nitric.yaml` file with an entry for the function
```
Comment (tjholm) [inline docs/reference/project/functions.mdx:18]: Not currently relevant (this functionality has been removed from the next version of the SDK

**Review by tjholm — CHANGES_REQUESTED** [review]

File: docs/reference/project/functions.mdx [hunk 2]
```diff
@@ -1 +1,30 @@
...
+
+## Properties
+
+`name` the name of the Nitric function. Must be unique to the stack file as it is used as the reference to the function for entrypoints and apis.
+
+`handler` the path to the function handler, relative to the root of the project.
+
+There are optional properties for when triggering a function with a topic.
+
+`triggers` the [topic(s)](./topics.mdx) that this function subscribes to.
```
Comment (tjholm) [inline docs/reference/project/functions.mdx:12]: For other reference docs I've used a table to describe properties. Nothing wrong with this format, but we should pick one to maintain consistency.

Comment (HomelessDinosaur) [reply docs/reference/project/functions.mdx:12]: I will change it to the table

File: docs/reference/project/functions.mdx [hunk 3]
```diff
@@ -1 +1,28 @@
-TODO...
+A Nitric function is described by its definition in a `nitric.yaml` file. Nitric functions are a type of [Service](../../intro/services) that offers a portable way to develop function-as-a-service (FaaS) applications, which can then be deployed to
+popular Serverless offerings like AWS Lamda, Google CloudRun and Azure AppService.
+
+## Properties
+
+| Property | Value      | Required | Description                                                                                                                                     |
+| -------- | ---------- | -------- | ----------------------------------------------------------------------------------------------------------------------------------------------- |
+| name     | `string`   | Yes      | The name of the Nitric function. Must be unique to the stack file as it is used as the reference to the function for entrypoints and apis.      |
```
Comment (tjholm) [inline docs/reference/project/functions.mdx:8]: Looks good, one more catch with `name` as a property, is it isn't really a property of a function and more part of it's general definition, as it's just a key that a function object is stored under. I've been trying to think of a way to represent this and listing it as a property could work. Just a note that it's not in other property tables.

Comment (HomelessDinosaur) [reply docs/reference/project/functions.mdx]: I'll remove it for consistency amongst the reference docs.

**Review by tjholm — APPROVED** [review]

**Pull request merged**